chia/util/file_keyring.py

```diff
@@ -112,69 +109,124 @@ def inner(self: FileKeyring, *args: object, **kwargs: object) -> Optional[str]:
     return inner
 
 
+class FileKeyringVersionError(Exception):
+    def __int__(self, actual_version: int) -> None:
+        super().__init__(
+            f"Keyring format is unrecognized. Found version {actual_version}"
+            f", expected a value <= {MAX_SUPPORTED_VERSION}. "
+            "Please update to a newer version"
+        )
+
+
 @final
 @dataclass
-class FileKeyring(FileSystemEventHandler):  # type: ignore[misc] # Class cannot subclass "" (has type "Any")
+class FileKeyringContent:
+    """
+    FileKeyringContent represents the data structure of the keyring file. It contains an encrypted data part which is
+    encrypted with a key derived from the user-provided master passphrase.
     """
-    FileKeyring provides an file-based keyring store that is encrypted to a key derived
-    from the user-provided master passphrase. The public interface is intended to align
-    with the API provided by the keyring module such that the KeyringWrapper class can
-    pick an appropriate keyring store backend based on the OS.
 
-    The keyring file format uses YAML with a few top-level keys:
+    # The version of the whole keyring file structure
+    version: int = 1
+    # Random salt used as a PBKDF2 parameter. Updated when the master passphrase changes
+    salt: bytes = field(default_factory=generate_salt)
+    # Random nonce used as a ChaCha20Poly1305 parameter. Updated on each write to the file.
+    nonce: bytes = field(default_factory=generate_nonce)
+    # Encrypted and base64 decoded keyring data.
```

Comment:
```suggestion
    # Encrypted and base64 encoded keyring data.
```

Reply:
I'd mention that the data is prepended with the checkbytes value, even though the checkbytes are added/removed elsewhere.